oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/dsl/SampleFamily.java

```diff
@@ -376,15 +374,63 @@ public SampleFamily endpoint(List<String> serviceKeys, List<String> endpointKeys
         if (this == EMPTY) {
             return EMPTY;
         }
-        this.context.setMeterEntity(MeterEntity.newEndpoint(dim(serviceKeys), dim(endpointKeys)));
-        return left(io.vavr.collection.Stream.concat(serviceKeys, endpointKeys).asJava());
+        return createMeterSamples(serviceKeys, endpointKeys, ScopeType.ENDPOINT);
+    }
+
+    private SampleFamily createMeterSamples(List<String> serviceKeys, List<String> level2keys, ScopeType scopeType) {
+        final List<String> labelKeys = io.vavr.collection.Stream.concat(serviceKeys, level2keys).asJava();
```

Comment:
`o.vavr.collection.Stream` Full class name?

Reply:
This `final List<String> labelKeys = io.vavr.collection.Stream.concat(serviceKeys, level2keys).asJava()` should be provided inside EntityDescription interface.

Reply:
down